src/pip/_internal/network/download.py

```diff
@@ -170,7 +165,7 @@ def __call__(
                 logger.critical(
                     "HTTP error %s while getting %s",
                     e.response.status_code,
-                    link,
+                    link.show_url,
```

Comment:
This should not be show_url but the entire URL.

Reply:
So its still inconsistent link repr in logs - as described in the issue